Compilation issues with gdc - error: unrecognized command-line option ‘-i=sokol’

```
user@debian:~/aaa/foo$ gdc-15 --version
gdc-15 (Debian 15.1.0-5) 15.1.0
```


```
user@debian:~/aaa/foo$ dub build --compiler=gdc-15
    Starting Performing "debug" build using gdc-15 for x86_64.
    Building sokol-d 0.3.2: building configuration [sokol-static]
   Pre-build Running commands
In file included from /home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_app.c:5:
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_app.h: In function ‘_sapp_x11_get_clipboard_string’:
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_app.h:10728:16: warning: missing initializer for field ‘revents’ of ‘struct pollfd’ [-Wmissing-field-initializers]
10728 |         struct pollfd fd = { ConnectionNumber(_sapp.x11.display), POLLIN };
      |                ^~~~~~
In file included from /usr/include/poll.h:1,
                 from /home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_app.h:2279:
/usr/include/x86_64-linux-gnu/sys/poll.h:40:15: note: ‘revents’ declared here
   40 |     short int revents;          /* Types of events that actually occurred.  */
      |               ^~~~~~~
In file included from /home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.c:6:
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h: In function ‘sshape_plane_sizes’:
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:853:5: warning: missing initializer for field ‘indices’ of ‘sshape_sizes_t’ [-Wmissing-field-initializers]
  853 |     sshape_sizes_t res = { {0} };
      |     ^~~~~~~~~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:434:25: note: ‘indices’ declared here
  434 |     sshape_sizes_item_t indices;
      |                         ^~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h: In function ‘sshape_box_sizes’:
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:863:5: warning: missing initializer for field ‘indices’ of ‘sshape_sizes_t’ [-Wmissing-field-initializers]
  863 |     sshape_sizes_t res = { {0} };
      |     ^~~~~~~~~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:434:25: note: ‘indices’ declared here
  434 |     sshape_sizes_item_t indices;
      |                         ^~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h: In function ‘sshape_sphere_sizes’:
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:873:5: warning: missing initializer for field ‘indices’ of ‘sshape_sizes_t’ [-Wmissing-field-initializers]
  873 |     sshape_sizes_t res = { {0} };
      |     ^~~~~~~~~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:434:25: note: ‘indices’ declared here
  434 |     sshape_sizes_item_t indices;
      |                         ^~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h: In function ‘sshape_cylinder_sizes’:
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:883:5: warning: missing initializer for field ‘indices’ of ‘sshape_sizes_t’ [-Wmissing-field-initializers]
  883 |     sshape_sizes_t res = { {0} };
      |     ^~~~~~~~~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:434:25: note: ‘indices’ declared here
  434 |     sshape_sizes_item_t indices;
      |                         ^~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h: In function ‘sshape_torus_sizes’:
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:893:5: warning: missing initializer for field ‘indices’ of ‘sshape_sizes_t’ [-Wmissing-field-initializers]
  893 |     sshape_sizes_t res = { {0} };
      |     ^~~~~~~~~~~~~~
/home/user/.dub/packages/sokol-d/0.3.2/sokol-d/src/sokol/c/sokol_shape.h:434:25: note: ‘indices’ declared here
  434 |     sshape_sizes_item_t indices;
      |                         ^~~~~~~
gdc-15: error: unrecognized command-line option ‘-i=sokol’
gdc-15: error: unrecognized command-line option ‘-i=shaders’
gdc-15: error: unrecognized command-line option ‘-i=handmade’
Error gdc-15 failed with exit code 1.
```



dub from Debian repos, 
```
user@debian:~/aaa/foo$ dub --version
DUB version 1.40.0-1, built on May  3 2025
```


Might want to check https://gcc.gnu.org/onlinedocs/gcc-15.1.0/gdc/Directory-Options.html and https://gcc.gnu.org/onlinedocs/gcc-15.1.0/gdc/ImportC.html

This probably should be fixed on a dub side maybe. (It is unlikely to be done in gcc due for foreseeable feature), or handled in the dub.json somehow.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -112,12 +112,10 @@ jobs:
           dub build :userdata -c wasm -v --arch wasm32-unknown-emscripten-wasm
           dub build :vertexpull -c wasm -v --arch wasm32-unknown-emscripten-wasm
 
-  gdc-15:
+  archlinux/gdc-15:
     strategy:
       fail-fast: false
-      matrix:
-        runs-on: [ubuntu-latest]
-    runs-on: ${{ matrix.runs-on }}
+    runs-on: ubuntu-latest
     container:
       image: archlinux:latest
     defaults:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -116,12 +116,19 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        runs-on: [ubuntu-24.04-arm, ubuntu-latest]
+        runs-on: [ubuntu-latest]
     runs-on: ${{ matrix.runs-on }}
+    container:
+      image: archlinux:latest
+    defaults:
+      run:
+        shell: bash
     steps:
       - uses: actions/checkout@v4
-      - name: install gdc
-        run: sudo apt-get update && sudo apt-get install gdc dub
+      - name: prepare-linux
+        run: |
+            pacman -Syu --noconfirm
+            pacman -S --noconfirm glu mesa xorg alsa-lib gdc dub      
       - name: (dub) Build sokol examples
         working-directory: examples
         run: |

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -112,7 +112,7 @@ jobs:
           dub build :userdata -c wasm -v --arch wasm32-unknown-emscripten-wasm
           dub build :vertexpull -c wasm -v --arch wasm32-unknown-emscripten-wasm
 
-  archlinux/gdc-15:
+  archlinux-gdc-15:
     strategy:
       fail-fast: false
     runs-on: ubuntu-latest

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -112,7 +112,7 @@ jobs:
           dub build :userdata -c wasm -v --arch wasm32-unknown-emscripten-wasm
           dub build :vertexpull -c wasm -v --arch wasm32-unknown-emscripten-wasm
 
-  gdc:
+  gdc-15:
     strategy:
       fail-fast: false
       matrix:
@@ -128,7 +128,7 @@ jobs:
       - name: prepare-linux
         run: |
             pacman -Syu --noconfirm
-            pacman -S --noconfirm glu mesa xorg alsa-lib gdc dub      
+            pacman -S --noconfirm glu mesa xorg alsa-lib gcc gdc dub      
       - name: (dub) Build sokol examples
         working-directory: examples
         run: |

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -121,7 +121,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - name: install gdc
-        run: sudo apt-get update && sudo apt-get install gdc
+        run: sudo apt-get update && sudo apt-get install gdc dub
       - name: (dub) Build sokol examples
         working-directory: examples
         run: |

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -137,8 +137,6 @@ jobs:
           dub build :clear
           dub build :cube
           dub build :debugtext
-          dub build :droptest
-          dub build :imgui
           dub build :instancing
           dub build :instancingcompute
           dub build :mrt

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -112,3 +112,37 @@ jobs:
           dub build :userdata -c wasm -v --arch wasm32-unknown-emscripten-wasm
           dub build :vertexpull -c wasm -v --arch wasm32-unknown-emscripten-wasm
 
+  gdc:
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-24.04-arm, ubuntu-latest]
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: install gdc
+        run: sudo apt-get update && sudo apt-get install gdc
+      - name: (dub) Build sokol examples
+        working-directory: examples
+        run: |
+          dub build :blend -f
+          dub build :bufferoffsets
+          dub build :clear
+          dub build :cube
+          dub build :debugtext
+          dub build :droptest
+          dub build :imgui
+          dub build :instancing
+          dub build :instancingcompute
+          dub build :mrt
+          dub build :noninterleaved
+          dub build :offscreen
+          dub build :quad
+          dub build :shapes
+          dub build :saudio
+          dub build :sglcontext
+          dub build :sglpoints
+          dub build :triangle
+          dub build :texcube
+          dub build :userdata
+          dub build :vertexpull

--- a/dub.json
+++ b/dub.json
@@ -75,12 +75,15 @@
 				"src/imgui/*.d",
 				"src/imgui/*.c"
 			],
-			"dflags": [
+			"dflags-dmd": [
 				"-i=sokol",
 				"-i=shaders",
 				"-i=handmade"
 			],
 			"dflags-ldc": [
+				"-i=sokol",
+				"-i=shaders",
+				"-i=handmade",
 				"-mcpu=native"
 			],
 			"targetName": "sokold",
@@ -172,12 +175,15 @@
 				"src/imgui/*.d",
 				"src/imgui/*.c"
 			],
-			"dflags": [
+			"dflags-dmd": [
 				"-i=sokol",
 				"-i=shaders",
 				"-i=handmade"
 			],
 			"dflags-ldc": [
+				"-i=sokol",
+				"-i=shaders",
+				"-i=handmade",
 				"-mcpu=native",
 				"--link-defaultlib-shared=false"
 			],
@@ -271,12 +277,15 @@
 				"src/imgui/*.d",
 				"src/imgui/*.c"
 			],
-			"dflags": [
+			"dflags-dmd": [
 				"-i=sokol",
 				"-i=shaders",
 				"-i=handmade"
 			],
 			"dflags-ldc": [
+				"-i=sokol",
+				"-i=shaders",
+				"-i=handmade",
 				"-mcpu=native"
 			],
 			"targetName": "sokold",
@@ -315,12 +324,15 @@
 				"src/imgui/*.d",
 				"src/imgui/*.c"
 			],
-			"dflags": [
+			"dflags-dmd": [
 				"-i=sokol",
 				"-i=shaders",
 				"-i=handmade"
 			],
 			"dflags-ldc": [
+				"-i=sokol",
+				"-i=shaders",
+				"-i=handmade",
 				"-mcpu=native"
 			],
 			"targetType": "dynamicLibrary",
@@ -360,21 +372,22 @@
 			"versions": [
 				"has_imgui"
 			],
-			"dflags": [
+			"dflags-dmd": [
 				"-i=sokol",
 				"-i=shaders",
 				"-i=handmade",
-				"-i=imgui"
-			],
-			"dflags-dmd": [
+				"-i=imgui",
 				"-P-I$PACKAGE_DIR/vendor/imgui/src"
 			],
 			"dflags-gdc": [
 				"-I$PACKAGE_DIR/vendor/imgui/src"
 			],
 			"dflags-ldc": [
+				"-i=sokol",
+				"-i=shaders",
+				"-i=handmade",
+				"-i=imgui",
 				"-P-I$PACKAGE_DIR/vendor/imgui/src",
-				"-Xcc=-std=c99",
 				"-mcpu=native"
 			],
 			"libs": [
@@ -499,21 +512,22 @@
 			"versions": [
 				"has_imgui"
 			],
-			"dflags": [
+			"dflags-dmd": [
 				"-i=sokol",
 				"-i=shaders",
 				"-i=handmade",
-				"-i=imgui"
-			],
-			"dflags-dmd": [
+				"-i=imgui",
 				"-P-I$PACKAGE_DIR/vendor/imgui/src"
 			],
 			"dflags-gdc": [
 				"-I$PACKAGE_DIR/vendor/imgui/src"
 			],
 			"dflags-ldc": [
+				"-i=sokol",
+				"-i=shaders",
+				"-i=handmade",
+				"-i=imgui",
 				"-P-I$PACKAGE_DIR/vendor/imgui/src",
-				"-Xcc=-std=c99",
 				"-mcpu=native",
 				"--link-defaultlib-shared=false"
 			],

--- a/dub.json
+++ b/dub.json
@@ -2,7 +2,7 @@
 	"name": "sokol-d",
 	"description": "D bindings for sokol",
 	"license": "Zlib",
-	"version": "0.3.2",
+	"version": "0.3.3",
 	"authors": [
 		"Matheus França",
 		"Andre Weissflog"

--- a/dub.json
+++ b/dub.json
@@ -113,7 +113,9 @@
 				"GL",
 				"Xi",
 				"Xcursor",
-				"asound"
+				"asound",
+				"m",
+				"dl"
 			],
 			"libs-windows": [
 				"dxgi",
@@ -215,7 +217,9 @@
 				"GL",
 				"Xi",
 				"Xcursor",
-				"asound"
+				"asound",
+				"m",
+				"dl"
 			],
 			"libs-windows": [
 				"dxgi",
@@ -298,7 +302,9 @@
 				"wayland-egl",
 				"wayland-cursor",
 				"xkbcommon",
-				"asound"
+				"asound",
+				"m",
+				"dl"
 			],
 			"preBuildCommands-linux": [
 				"mkdir -p $PACKAGE_DIR/build",
@@ -346,7 +352,9 @@
 				"wayland-egl",
 				"wayland-cursor",
 				"xkbcommon",
-				"asound"
+				"asound",
+				"m",
+				"dl"
 			],
 			"preBuildCommands-linux": [
 				"mkdir -p $PACKAGE_DIR/build",
@@ -400,7 +408,9 @@
 				"GL",
 				"Xi",
 				"Xcursor",
-				"asound"
+				"asound",
+				"m",
+				"dl"
 			],
 			"libs-osx": [
 				"c++",
@@ -541,7 +551,9 @@
 				"GL",
 				"Xi",
 				"Xcursor",
-				"asound"
+				"asound",
+				"m",
+				"dl"
 			],
 			"libs-osx": [
 				"c++",

--- a/examples/blend/dub.json
+++ b/examples/blend/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "blend",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/bufferoffsets/dub.json
+++ b/examples/bufferoffsets/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "bufferoffsets",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/clear/dub.json
+++ b/examples/clear/dub.json
@@ -10,6 +10,38 @@
     },
     "targetPath": "build",
     "targetName": "clear",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/cube/dub.json
+++ b/examples/cube/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "cube",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/debugtext/dub.json
+++ b/examples/debugtext/dub.json
@@ -10,6 +10,38 @@
     },
     "targetPath": "build",
     "targetName": "debugtext",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/droptest/dub.json
+++ b/examples/droptest/dub.json
@@ -10,6 +10,38 @@
     },
     "targetPath": "build",
     "targetName": "droptest",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/dub.json
+++ b/examples/dub.json
@@ -6,38 +6,6 @@
 		"Matheus França",
 		"Andre Weissflog"
 	],
-	"buildTypes": {
-		"debug": {
-			"buildOptions": [
-				"debugMode",
-				"debugInfo"
-			]
-		},
-		"release": {
-			"buildOptions": [
-				"releaseMode",
-				"inline",
-				"optimize"
-			]
-		},
-		"release-nobounds": {
-			"buildOptions": [
-				"releaseMode",
-				"inline",
-				"noBoundsCheck",
-				"optimize"
-			]
-		},
-		"release-betterc": {
-			"buildOptions": [
-				"betterC",
-				"releaseMode",
-				"inline",
-				"noBoundsCheck",
-				"optimize"
-			]
-		}
-	},
 	"subPackages": [
 		"blend",
 		"bufferoffsets",

--- a/examples/imgui/dub.json
+++ b/examples/imgui/dub.json
@@ -10,6 +10,38 @@
     },
     "targetPath": "build",
     "targetName": "imgui",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/instancing/dub.json
+++ b/examples/instancing/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "instancing",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/instancingcompute/dub.json
+++ b/examples/instancingcompute/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "instancingcompute",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/mrt/dub.json
+++ b/examples/mrt/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "mrt",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/noninterleaved/dub.json
+++ b/examples/noninterleaved/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "noninterleaved",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/offscreen/dub.json
+++ b/examples/offscreen/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "offscreen",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/quad/dub.json
+++ b/examples/quad/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "quad",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/saudio/dub.json
+++ b/examples/saudio/dub.json
@@ -10,6 +10,38 @@
     },
     "targetPath": "build",
     "targetName": "saudio",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/sglcontext/dub.json
+++ b/examples/sglcontext/dub.json
@@ -10,6 +10,38 @@
     },
     "targetPath": "build",
     "targetName": "sglcontext",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/sglpoints/dub.json
+++ b/examples/sglpoints/dub.json
@@ -10,6 +10,38 @@
     },
     "targetPath": "build",
     "targetName": "sglpoints",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/shapes/dub.json
+++ b/examples/shapes/dub.json
@@ -13,9 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "shapes",
-    "buildOptions": [
-        "betterC"
-    ],
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/texcube/dub.json
+++ b/examples/texcube/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "texcube",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/triangle/dub.json
+++ b/examples/triangle/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "triangle",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/userdata/dub.json
+++ b/examples/userdata/dub.json
@@ -10,6 +10,38 @@
     },
     "targetPath": "build",
     "targetName": "userdata",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

--- a/examples/vertexpull/dub.json
+++ b/examples/vertexpull/dub.json
@@ -13,6 +13,38 @@
     },
     "targetPath": "build",
     "targetName": "vertexpull",
+    "buildTypes": {
+        "debug": {
+            "buildOptions": [
+                "debugMode",
+                "debugInfo"
+            ]
+        },
+        "release": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "optimize"
+            ]
+        },
+        "release-nobounds": {
+            "buildOptions": [
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        },
+        "release-betterc": {
+            "buildOptions": [
+                "betterC",
+                "releaseMode",
+                "inline",
+                "noBoundsCheck",
+                "optimize"
+            ]
+        }
+    },
     "configurations": [
         {
             "name": "native",

